src/Microsoft.OpenApi/Reader/OpenApiReaderRegistry.cs

```diff
@@ -12,7 +13,7 @@ namespace Microsoft.OpenApi.Reader
     /// </summary>
     public static class OpenApiReaderRegistry
     {
-        private static readonly Dictionary<string, IOpenApiReader> _readers = new(StringComparer.OrdinalIgnoreCase);
+        private static readonly ConcurrentDictionary<string, IOpenApiReader> _readers = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
line 30 should also be probably updated to `_readers.AddOrUpdate(format,reader, (_,_) => reader);`